.github/workflows/ci.yml

```diff
@@ -0,0 +1,35 @@
+name: CI
+
+# On every pull request, but only on push to master
+on:
+  push:
+    branches:
+    - master
```

Comment:
Maybe also add `maint*` to run on all maintenance branches (like [maint-1.2](https://github.com/rasterio/rasterio/tree/maint-1.2)).